source/monitoring/bucket-notifications/publish-events-to-kafka.rst

```diff
@@ -0,0 +1,377 @@
+.. _minio-bucket-notifications-publish-kafka:
+
+=======================
+Publish Events to Kafka
+=======================
+
+.. default-domain:: minio
+
+.. contents:: Table of Contents
+   :local:
+   :depth: 1
+
+MinIO supports publishing :ref:`bucket notification
+<minio-bucket-notifications>` events to Kafka. 
+
+MinIO relies on the :github:`Shopify/sarama` project for Kafka connectivity
+and shares that project's Kafka support. See the 
+``sarama`` :github:`Compatibility and API stability 
+<Shopify/sarama/#compatibility-and-api-stability>` section for more details. 
+MinIO requires at *minimum* Kafka 0.9.0 or 0.10.0.
```

Comment:
@vadmeste @poornas So this is mostly taken from the existing docs, and it doesn't make much sense to me. We state we *require* 0.9 and 0.10, but the Go library referenced here is only guaranteeing support for the two most recent major releases (2.7.X I think).

Reply:
> but the Go library referenced here is only guaranteeing support for the two most recent major releases (2.7.X I think).

Could you find where did you find this information ?

Reply:
https://github.com/Shopify/sarama#compatibility-and-api-stability

> Sarama provides a "2 releases + 2 months" compatibility guarantee: we support the two latest stable releases of Kafka and Go, and we provide a two month grace period for older releases. This means we currently officially support Go 1.14 through 1.16, and Kafka 2.6 through 2.7, although older releases are still likely to work

Not sure how tight our 0.9/.10 requirement is.

Reply:
cc @Praveenrajmani @kannappanr on the above, as you have some recent-ish commits for Kafka

Reply:
Can we not assure this - `MinIO requires at *minimum* Kafka 0.9.0 or 0.10.0.` without proper testing?

Reply:
@Praveenrajmani no idea - this is based on what we currently have documented [here](https://docs.min.io/docs/minio-bucket-notification-guide.html)

> MinIO requires Kafka version 0.10 or 0.9. Internally MinIO uses the Shopify/sarama library and so has the same version compatibility as provided by this library.

I'm *guessing* we just never went back to update those version minimums. The last commit to that section was 2 years ago, and it's unrelated beyond linking things. So it's probably been 2+ years.

If we can safely assume MinIO supports the versions the `sarama` library supports, then we can safely drop the 0.9.0/0.10.0 suggestion.

I wasn't able to test this myself, just based on time constraints - this is unfortunately an "import" pass, focusing on formatting and chasing down areas of clarity. Fully testing each of the notification targets would significantly increase the time required to complete any one of the target pages.